packages/core/src/tools/mcp-tool.ts

```diff
@@ -307,6 +313,57 @@ function transformResourceLinkBlock(block: McpResourceLinkBlock): Part {
   };
 }
 
+/**
+ * Sanitizes the MCP content blocks, replacing unsupported mime types
+ * with an error message. If no sanitization is needed, the original
+ * response is returned. Otherwise, a sanitized copy is returned.
+ * @param sdkResponse The raw Part[] array from `mcpTool.callTool()`.
+ * @returns A sanitized Part[] array.
+ */
+export function returnSanitizedMcpContent(sdkResponse: Part[]): Part[] {
+  let sanitizedResponse: Part[] | null = null;
+
+  for (let partIndex = 0; partIndex < sdkResponse.length; partIndex++) {
+    const funcResponse = sdkResponse[partIndex]?.functionResponse;
+    const mcpContent = funcResponse?.response?.['content'] as McpContentBlock[];
+
+    if (!Array.isArray(mcpContent)) {
+      continue;
+    }
+
+    for (let blockIndex = 0; blockIndex < mcpContent.length; blockIndex++) {
+      const block = mcpContent[blockIndex];
+      let mimeType: string | undefined;
+
+      if (block.type === 'image' || block.type === 'audio') {
+        mimeType = block.mimeType;
+      } else if (block.type === 'resource' && block.resource.blob) {
```

Comment:
I believe resource-link types can also have a mime-type fwiw

Reply:
Ok, if this ends up just falling back on a text representation that is fine for now, definitely some future work to be done for these though but it seems unrelated to this change.